### PR TITLE
Multiple correct output Feature 

### LINKED ISSUE #447: [Multiple correct outputs] Modify model for RegressionTestOutput
Modify https://github.com/CCExtractor/sample-platform/blob/master/mod_regression/models.py#L135 so that for a single generated file, multiple correct hashes can be entered. This will likely result in another table with file hashes that link to the id of the RegressionTestOutput.

### LINKED ISSUE #448: [Multiple correct outputs] Update GUI to allow adding a new correct value
Modify the test result view to add a link/button to update the list of correct files.

https://github.com/CCExtractor/sample-platform/blob/master/mod_test/controllers.py
https://github.com/CCExtractor/sample-platform/blob/master/templates/test/by_id.html

### LINKED ISSUE #449: [Multiple correct outputs] Modify pass/fail decision mechanism
The mechanism that decides if a test is passed/failed (https://github.com/CCExtractor/sample-platform/blob/master/mod_test/controllers.py#L130 and further) needs to be modified so that a test passes if the resulting file is one of the approved files.

### PR DESCRIPTION
**Pull request by kvshravan**
**[FEATURE]** 

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

---
fixes #447 
fixes #448 
fixes #449 
The following set of features were added :
1. Added multiple correct output feature, the platform can now detect multiple correct output files using RegressionTestOutputFiles table.
2. GUI to add and remove multiple correct outputs in regression test view.
3. Correct output can be chosen from  got values (if they're not None) of last 10 Test Runs.
4. Pass/Fail mechanism has also been modified so that a test is passed when the output file is one of the multiple correct files.

### CODE REVIEW

File: mod_regression/controllers.py [hunk 1]
```diff
@@ -195,14 +198,22 @@ def toggle_active_status(regression_id):
...
-def test_result_file(regression_test_output_id):
+@mod_regression.route('/test/<regression_test_output_id>/download/<multiple_files>', methods=['GET'])
+def test_result_file(regression_test_output_id, multiple_files):
     """View the output files of the regression test."""
-    rto = RegressionTestOutput.query.filter(RegressionTestOutput.id == regression_test_output_id).first()
-    if rto is None:
-        g.log.error(f'requested regression test output with id: {regression_test_output_id} not found!')
-        abort(404)
-    return serve_file_download(rto.filename_correct, 'TestResults', 'regression-download')
+    if multiple_files == 'True':
```
Comment (thealphadollar) [inline mod_regression/controllers.py:204]: IMO, should we make `multiple_files` an optional field to ensure backward compatibility. Also, I'm not very sure if the exact 'True' should be kept as the parameter.

cc/ @canihavesomecoffee

File: mod_regression/controllers.py [hunk 2]
```diff
@@ -195,14 +198,22 @@ def toggle_active_status(regression_id):
...
+        rtof = RegressionTestOutputFiles.query.filter(RegressionTestOutputFiles.id == regression_test_output_id).first()
+        print(rtof)
```
Comment (thealphadollar) [inline mod_regression/controllers.py:206]: Is this for debugging?

File: mod_regression/models.py [hunk 3]
```diff
@@ -206,3 +208,38 @@ def create_correct_filename(self, name) -> str:
         :rtype: str
         """
         return f"{name}{self.correct_extension}"
+
+
+class RegressionTestOutputFiles(Base):
+    """docstring for RegressionTestOutputFiles."""
```
Comment (thealphadollar) [inline mod_regression/models.py:214]: Can we display brief information about the class in the docstring?

**Review by thealphadollar — COMMENTED** [review]
Hey @kvshravan , thanks for the contribution. I really like how you have used modular code and segregated tasks into different functions.

I've left some comments. I see you have added a few new templates. I'd be interested to see how they look before I approve the PR.

File: mod_regression/controllers.py (hunk 1 above)
Comment (canihavesomecoffee) [reply mod_regression/controllers.py:204]: I'm not so sure why you'd want to keep that optional... It's not like we need to care about broken links for these downloads. As long as all calls inside our program are updated to use it, it should be fine...

**Review by canihavesomecoffee — CHANGES_REQUESTED** [review]
If you could address the things mentioned by @thealphadollar  where I gave a thumbs-up on, it looks good to me to merge. Nice work :+1:

File: mod_regression/controllers.py (hunk 1 above)
Comment (thealphadollar) [reply mod_regression/controllers.py:204]: Yeah. If we are simultaneously updating the calls, then good.

Comment (canihavesomecoffee) [reply mod_regression/controllers.py:204]: Well, at least one of the uses is further down in this PR :)

Let's check if there's others.

Comment (canihavesomecoffee) [reply mod_regression/controllers.py:204]: @kvshravan Can you make sure that this change doesn't break https://github.com/CCExtractor/sample-platform/blob/master/tests/test_regression/TestControllers.py#L52 (or that this test is adapted)?

**Comment by kvshravan** [comment]
@thealphadollar @canihavesomecoffee  I've segregated the code to download variant files , that should resolve any conflicts for the original code. For your reference I'm attaching two images how the templates look like.
Screenshot from 2021-02-12 13-40-47
Screenshot from 2021-02-12 13-40-40
Screenshot from 2021-02-12 13-40-27

**Review by canihavesomecoffee — APPROVED** [review]
LGTM. @thealphadollar is it ok for you?

**Pull request merged**